library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionAdapter.kt

```diff
@@ -0,0 +1,117 @@
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.chuckerteam.chucker.R
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import com.chuckerteam.chucker.internal.data.entity.HttpTransactionTuple
+import java.text.DateFormat
+
+internal class TransactionAdapter internal constructor(
+    context: Context,
+    private val listener: TransactionClickListListener?
+) : RecyclerView.Adapter<TransactionAdapter.ViewHolder>() {
+    private var transactions: List<HttpTransactionTuple> = arrayListOf()
+
+    private val colorDefault: Int = ContextCompat.getColor(context, R.color.chucker_status_default)
+    private val colorRequested: Int = ContextCompat.getColor(context, R.color.chucker_status_requested)
+    private val colorError: Int = ContextCompat.getColor(context, R.color.chucker_status_error)
+    private val color500: Int = ContextCompat.getColor(context, R.color.chucker_status_500)
+    private val color400: Int = ContextCompat.getColor(context, R.color.chucker_status_400)
+    private val color300: Int = ContextCompat.getColor(context, R.color.chucker_status_300)
+
+    override fun getItemCount(): Int = transactions.size
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val itemView =
+            LayoutInflater.from(parent.context)
+                .inflate(R.layout.chucker_list_item_transaction, parent, false)
+        return ViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
+        holder.bind(transactions[position])
+
+    fun setData(httpTransactions: List<HttpTransactionTuple>) {
+        this.transactions = httpTransactions
+        notifyDataSetChanged()
+    }
+
+    inner class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
+        private val code: TextView = view.findViewById(R.id.chucker_code)
+        private val path: TextView = view.findViewById(R.id.chucker_path)
+        private val host: TextView = view.findViewById(R.id.chucker_host)
+        private val start: TextView = view.findViewById(R.id.chucker_time_start)
+        private val duration: TextView = view.findViewById(R.id.chucker_duration)
+        private val size: TextView = view.findViewById(R.id.chucker_size)
+        private val ssl: ImageView = view.findViewById(R.id.chucker_ssl)
+
+        @SuppressLint("SetTextI18n")
+        fun bind(transaction: HttpTransactionTuple) {
+            path.text = "${transaction.method} ${transaction.path}"
+            host.text = transaction.host
+            start.text = DateFormat.getTimeInstance().format(transaction.requestDate)
+            ssl.visibility = if (transaction.isSsl) View.VISIBLE else View.GONE
```

Comment:
I was gonna say `ssl.isVisible = ...` and then I remembered that ktx is not a dependency of Chucker, to limit the size of the library :-D

Reply:
Yea, that's was exactly what I thought while I went over it